cmd/influx/setup.go

```diff
@@ -226,20 +226,25 @@ func onboardingRequest(ui *input.UI) (*influxdb.OnboardingRequest, error) {
 	}
 
 	req := &influxdb.OnboardingRequest{
-		User:            setupFlags.username,
-		Password:        setupFlags.password,
-		Org:             setupFlags.org,
-		Bucket:          setupFlags.bucket,
-		RetentionPeriod: influxdb.InfiniteRetention,
-		Token:           setupFlags.token,
+		User:                   setupFlags.username,
+		Password:               setupFlags.password,
+		Org:                    setupFlags.org,
+		Bucket:                 setupFlags.bucket,
+		RetentionPeriodSeconds: influxdb.InfiniteRetention,
+		Token:                  setupFlags.token,
 	}
 
 	if setupFlags.retention != "" {
 		dur, err := internal2.RawDurationToTimeDuration(setupFlags.retention)
 		if err != nil {
 			return nil, err
 		}
-		req.RetentionPeriod = dur
+		secs := dur / time.Second
```

Comment:
Would it be better here to use the Duration.Seconds() and Duration.Nanoseconds() to hide the internal representation in the Duration?  The modulo will have to change, if you do this, of course.